### ISSUE TITLE
Should Validation be set to false?

### ISSUE THREAD

**Issue by jbonzo** [mongodb/mongodb-enterprise-kubernetes#77]
I followed the tutorial to install Ops Manager as described [here](https://www.mongodb.com/blog/post/running-mongodb-ops-manager-in-kubernetes) but when I got to the step to apply the sample MongoDBOpsManager resource the command failed validation.

```
error: error validating "ops_manager/opsManager.yml": error validating data: 
[ValidationError(MongoDBOpsManager.spec.applicationDatabase): unknown field "podSpec"
in com.mongodb.v1.MongoDBOpsManager.spec.applicationDatabase, 
ValidationError(MongoDBOpsManager.spec): unknown field "configuration" in 
com.mongodb.v1.MongoDBOpsManager.spec]; if you choose to ignore these errors, turn 
validation off with --validate=false
```

It worked if I turned validation off, but I didn't see any mention of doing so in the tutorial or documentation. I was wondering if this should be added to the documentation or if there is a configuration error on my side.

I am running Kubernetes 1.15.3

**Comment by antonlisovenko** [comment]
hi @jbonzo thanks for reporting this.
I think you've hit the strict validation introduced in Kubernetes 1.15 which requires CRD to validate **all** the fields provided in the custom resource config. Our assumption was that as our CRD uses `apiextensions.k8s.io/v1beta1` version this validation produces warnings only but doesn't block the creationn/update. We see this in our e2e clusters.

Sounds like this didn't work out in your case and yes - `--validate` was a good work around.

We've created an internal ticket to align CRD validation rules with existing config: CLOUDP-59347

**Comment by jbonzo** [comment]
Thanks, I appreciate for the prompt followup.

**Comment by antonlisovenko** [comment]
hi @jbonzo 

out of curiosity: could you tell more about your cluster? Is it some public cloud provider or some minikube/Kind cluster?
The thing is that our CRDs use version `apiextensions.k8s.io/v1beta1` for backward compatibility which prevents them from undergoing structural schema validation even for K8s clusters > 1.15. So we don't see the errors you've mentioned for any K8s versions.

**Comment by antonlisovenko** [comment]
This issue should be resolved in the last Operator (1.4.5)
